src/components/accessories/billFilterForm/BillFilterForm.tsx

```diff
@@ -0,0 +1,167 @@
+import { useFormik } from "formik";
+import get from "lodash.get";
+import has from "lodash.has";
+import React, { FC, useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { object, string } from "yup";
+import {
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../libraries/formDataHandling/functions";
+import { IState } from "../../../types";
+import AutocompleteField from "../autocompleteField/AutocompleteField";
+import DateField from "../dateField/DateField";
+import { initialFields } from "./consts";
+import { searchPatient } from "../../../state/patients/actions";
+
+import "./styles.scss";
+import { BillFilterProps, TValues } from "./types";
+import { PatientDTO } from "../../../generated";
+
+const BillFilterForm: FC<BillFilterProps> = ({
+  onSubmit,
+  className,
+  fields,
+}) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const [searchPatientParams, setSearchPatientParams] = useState({} as TValues);
+
+  const handlePatientSearchChange = (event: any, value: string) => {
+    setSearchPatientParams({
+      id: !isNaN(+value) ? value : "",
+      firstName: isNaN(+value) ? value : "",
+      secondName: "",
+      birthDate: "",
+      address: "",
+    });
+    dispatch(searchPatient(searchPatientParams));
+  };
+
```

Comment:
Actually, I don't understand what's the puropose of this. Why +value is taken as id and firstName?

Reply:
I removed it. The idea here was to let user find user by typing the patient code also.

Reply:
Ok maybe we can go though about. Maybe we can reuse this logic and instead of using a textfield we can filter the users by typing name or surname or code and use an autocomplete field in order to show them.

This sounds like a new Jira task for me, maby just poolish your code and then I open a new task.